lib/mongodb.js

```diff
@@ -67,7 +67,11 @@ exports.initialize = function initializeDataSource(dataSource, callback) {
   dataSource.ObjectID = mongodb.ObjectID;
 
   if (callback) {
-    dataSource.connector.connect(callback);
+    if (s.lazyConnect === "true") {
```

Comment:
Connectors should not need to know that the value was initially get from an ENV variable and is string. Since this is a boolean flag, the connector should expect a boolean value, not a string. 

It's the responsibility of the code reading the env variable (e.g. in loopback-boot) to convert the variable into the appropriate javascript type (`boolean`). 

This check then becomes `if (s.lazyConnect)`.